modules/components/pages/outputs/snowflake_streaming.adoc

```diff
@@ -0,0 +1,520 @@
+          _ => "VARIANT"
+        }
+    build_parallelism: 1
+    batching:
+      count: 0
+      byte_size: 0
+      period: ""
+      check: ""
+      processors: [] # No default (optional)
+    max_in_flight: 64
+    channel_prefix: "" # No default (optional)
+```
+
+--
+======
+
+`<table-name>`: Replace the placeholder value with the name of the Snowflake table you want to write data to.
+
+
+== Supported data formats for Snowflake columns
+
+The message data from your output must match the columns in the Snowflake that you want to write data to. The following table shows you the https://docs.snowflake.com/en/sql-reference/intro-summary-data-types[column data types supported by Snowflake^] and how they correspond to the xref:guides:bloblang/methods.adoc#type[Bloblang data types] in Redpanda Connect.
+
+|===
+| Snowflake column data type | Bloblang data types
+
+| CHAR, VARCHAR
+| `string`
+
+| BINARY
+| `string` or `bytes`
+
+| NUMBER 
+| `number`, or `string` where the `string` is parsed into a `number`
+
+| FLOAT 
+| `number`
+
+| BOOLEAN
+| `bool`, or `number` where a non-zero number is `true`
+
+| TIME, DATE, TIMESTAMP 
+| `timestamp`, or `number` where the `number` is a converted to a Unix timestamp
+
+| VARIANT, ARRAY, OBJECT
+| Any data type converted into JSON
+
+| GEOGRAPHY,GEOMETRY
+| Not supported
+
+|===
+
+
+== Authentication
+
+You can authenticate with Snowflake using an https://docs.snowflake.com/en/user-guide/key-pair-auth[RSA key pair^]. Either specify:
+
+* A PEM-encoded private key, in the <<private_key,`private_key` field>>.
+* The path to a file from which the output can load the private RSA key, in the <<private_key_file,`private_key_file` field>>.
+
+
+== Performance
+
+For improved performance, this output:
+
+* Sends multiple messages in parallel. You can tune the maximum number of in-flight messages (or message batches) with the field `max_in_flight`.
+* Sends messages as a batch. You can configure batches at both the input and output level. For more information, see xref:configuration:batching.adoc[Message Batching].
+
+=== Batch sizes
+
+Redpanda recommends that every message batch writes at least 16 MiB of compressed output to Snowflake. You can monitor batch sizes using the `snowflake_compressed_output_size_bytes` metric.
+
+=== Metrics
+
+This output emits the following metrics.
+
+|===
+| Metric name | Description
+
+| `snowflake_build_output_latency_ns`
+| The time taken to build the output file that is sent to Snowflake.
+
+| `snowflake_upload_latency_ns`
+| The time taken to upload the output file to Snowflake.
+
+| `snowflake_convert_latency_ns`
+| The time taken to convert messages into the Snowflake column data types.
+
+| `snowflake_serialize_latency_ns`
+| The time taken to serialize the converted columnar data into a file to send to snowflake.
+
+|===
+
+NOTE: `snowflake_convert_latency_ns` + `snowflake_serialize_latency_ns` = `snowflake_build_output_latency_ns`
```

Comment:
Maybe instead of the note we add to the description of `build_output_latency_ns` to mention it's the sum of the convert and serialize

Reply:
@rockwotj - I've added this to the table as suggested. I've also reordered the metrics into a sequence, which I hope is correct, and added in the `snowflake_compressed_output_size_bytes` metric.